src/SiteMaster/Core/Auditor/Site/Review/EditForm.php

```diff
@@ -0,0 +1,214 @@
+                throw new InvalidArgumentException('Could not find that review', 400);
+            }
+            
+            $this->site = $this->review->getSite();
+        } else {
+            if (!$this->site = Site::getByID($this->options['site_id'])) {
+                throw new InvalidArgumentException('Could not find that site', 400);
+            }
+        }
+        
+        $this->current_user = Session::getCurrentUser();
+
+        if (!$this->canEdit()) {
+            throw new AccessDeniedException('You do not have permission to edit this review.  You must be an admin.', 403);
+        }
+    }
+
+    /**
+     * Get the url for this page
+     *
+     * @return bool|string
+     */
+    public function getURL()
+    {
+        return $this->review->getURL() . 'edit/';
+    }
+
+    /**
+     * Get the title for this page
+     *
+     * @return string
+     */
+    public function getPageTitle()
+    {
+        return 'Edit Review';
+    }
+
+    /**
+     * Only Admin's can edit reviews
+     *
+     * @return bool
+     */
+    public function canEdit()
+    {
+        if (!$this->current_user) {
+            return false;
+        }
+
+        $review = $this->review;
+        
+        if (false == $review) {
+            $review = new Review();
+        }
+        
+        return $review->canEdit($this->current_user);
+    }
+
+    public function handlePost($get, $post, $files)
+    {
+        if (!isset($post['action'])) {
+            throw new InvalidArgumentException('An action must be specified', 400);
+        }
+
+        switch ($post['action']) {
+            case 'edit':
+                $this->edit($get, $post, $files);
+                break;
+            case 'delete':
+                $this->delete($get, $post, $files);
+                break;
+            default:
+                throw new InvalidArgumentException('An invalid action was given', 400);
+        }
+    }
+
+    /**
+     * handle the edit post action
+     *
+     * @param $get
+     * @param $post
+     * @param $files
+     */
+    protected function edit($get, $post, $files)
+    {
+        if (!isset($post['date_scheduled'])) {
+            throw new InvalidArgumentException('The review must be scheduled.', 400);
+        }
+        
+        if (!$this->review) {
+            $this->review = Review::createNewReview($this->site->id, $this->current_user->id, $post['date_scheduled']);
+        }
+        
+        $this->review->date_scheduled = $post['date_scheduled'];
+
+        if (isset($post['date_reviewed'])) {
+            $this->review->date_reviewed = Util::epochToDateTime(strtotime($post['date_reviewed']));
+        }
+
+        if (isset($post['internal_notes'])) {
+            $this->review->internal_notes = $post['internal_notes'];
+        }
+
+        if (isset($post['public_notes'])) {
+            $this->review->public_notes = $post['public_notes'];
+        }
+
+        if (isset($post['status'])) {
```

Comment:
Yes, something like that is good. Need to make sure there are no side-effects from parsing the input string (like if `strtotime` has an error/ returns false). How far to go with the validation depends on the input formats you'd like to support. The SO article provide _very_ strict formatting. Would you like to support submitted strings that `strtotime` can transform to dates automatically (ex: `"+5 days"`)?